### ISSUE TITLE
Rush - Support workspaces features 

### ISSUE THREAD

**Issue by sunilsurana** [microsoft/rushstack#1669]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [ ] Feature

An organisation can decide to host all of its products code under one repo. In this case when a developer starts to work on repo when doing rush install he will have to download whole npm dependencies which are not even related to his work. And this can get bad in PR builds.

**Feature ask**
Create a workspace feature where the rush.json structure looks like 
{
Worspace1:
       [package A, 
        package B], 
Workspace2:{
       [packageC,
        packageD]
}

So when someone run *rush install workspace1* it will only download dependencies for packages in that workspace.

**Comment by octogonz** [comment]
https://github.com/microsoft/rushstack/issues/1553 is a top priority on our roadmap for Rush.  It will delegate most of the installation responsibility to the package manager.  And then we'd try to solve this problem using a PNPM feature rather than a Rush-specific feature.

That said, I'm not sure that PNPM supports this today.  @sunilsurana could you investigate that, and if it's supported, could your report back with some details about how PNPM's feature works?

Your proposed feature sounds pretty valuable, so I'd like to get started on the design early in case it has any implications for https://github.com/microsoft/rushstack/issues/1553.

> So when someone run rush install workspace1 it will only download dependencies for packages in that workspace.

This would be complex for Rush to implement.  The package manager is in the best position to perform this sort of filtering.

**Comment by sunilsurana** [comment]
In Pnpm i see that there is option the specify filter and supply package using different patterns but not a standardized way to say install all dependecies for productA or something like that. 
Do you think it makes sense to create feature ask for this on PNPM?

**Comment by octogonz** [comment]
Yes. The basic idea would be to install a subset of the package-lock.json tree, based on a set of monorepo projects that we want to build. I feel like this should be fairly easy for PNPM to do within its current model. I bet people have asked it before. Please CC me on your issue so I can follow along. Thanks!

**Comment by DylanVann** [comment]
This will become easier with PnP / Yarn 2 since the package manager doesn't have to resolve conflicts between versions.

It would probably be a Yarn feature though. e.g. You could ask Yarn to install dependencies for a workspace (which could possibly also install them for workspaces it depends on).

**Comment by D4N14L** [comment]
I've created a separate issue to track PNPM workspaces support here: #1887
